### PR TITLE
Initial BUILD.gn files so that it can be used for Skia-Dawn

### PR DESCRIPTION
**Pull request by Kangz**
PTAL

### CODE REVIEW

File: build_overrides/build.gni [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# Don't use Chromium's third_party/binutils
+linux_use_bundled_binutils_override = false
+
+# In standalone Dawn builds, don't try to use the hermetic install of Xcode
+# that Chromium uses
+use_system_xcode = true
+
+# Android 32-bit non-component, non-clang builds cannot have symbol_level=2
+# due to 4GiB file size limit, see https://crbug.com/648948.
+# Set this flag to true to skip the assertion because we aren't on Android.
```
[the remark this answers is not in the document]
Comment (Kangz) [reply build_overrides/build.gni]: Even ANGLE that builds on Android has this set to true. I'll just remove the last line of the comment.

File: generator/templates/dawn_native/ProcTable.cpp [hunk 2]
```diff
@@ -55,6 +55,7 @@ namespace dawn_native {
                         (void) self;
                     {% endif %}
                     bool error = false;
+                    (void) error;
```
Comment (kainino0x) [inline generator/templates/dawn_native/ProcTable.cpp]: Didn't we have an NXT_UNUSED for this?

Comment (Kangz) [reply generator/templates/dawn_native/ProcTable.cpp]: Done

**Comment by Kangz** [comment]
Thanks all. @SenorBlanco, good luck with the rename 😉

**Pull request merged**